manager/src/main/java/com/redhat/service/bridge/manager/connectors/AbstractConnectorWorker.java

```diff
@@ -0,0 +1,127 @@
+        return connectorEntity;
+    }
+
+    protected abstract ConnectorEntity updateEntityForError(ConnectorEntity connectorEntity, Throwable error);
+
+    protected void afterSuccessfullyUpdated(ConnectorEntity c) {
+    }
+
+    // Used for workers that should delete the entity after successfully calling the service.
+    // By default, after the service call updateEntityForSuccess is called, with this physical deletion will occur
+    protected boolean shouldDeleteAfterSuccess() {
+        return false;
+    }
+
+    // Could be the k8s pod name https://issues.redhat.com/browse/MGDOBR-336
+    String workerId = UUID.randomUUID().toString();
+
+    // It's important that at the beginning there is no transaction active
+    // in this way we're sure each step in the following method is executed in its own transaction
+    @Transactional(Transactional.TxType.NEVER)
+    public void execute(ConnectorEntity currentEntity) {
+
+        ConnectorEntity connectorEntity;
+        try {
+            connectorEntity = claimConnector(currentEntity);
+            LOGGER.info("Running worker: {} on Entity: {} with workerId: {} ", workerName(), connectorEntity, workerId);
+        } catch (Exception e) {
+            // Cannot claim, return the unmodified entity
+            LOGGER.trace("Worker {} cannot claim Entity: {} with error: {} ", workerId, workerName(), e);
+            return;
+        }
+
+        try {
+            R serviceResponse = callExternalService(connectorEntity);
+
+            if (shouldDeleteAfterSuccess()) { // Used only for workers that physically delete entities
+                deleteEntity(connectorEntity);
+            } else { // Success path, most common
+                ConnectorEntity successConnectorEntity = successfullyCalledService(connectorEntity, serviceResponse);
+                afterSuccessfullyUpdated(successConnectorEntity);
+            }
+        } catch (Exception e) {
+
+            errorWhileCalling(e, connectorEntity);
+        }
+    }
+
+    @Transactional(Transactional.TxType.REQUIRES_NEW)
+    public ConnectorEntity claimConnector(ConnectorEntity currentEntity) {
+        // As this is a new transaction from a non-transactional state
+        // entity is in detached (non managed) state
+        // we need to merge it to flush the edit
+        ConnectorEntity mergedCE = entityManager.merge(currentEntity);
+        mergedCE.setWorkerId(workerId);
+
+        return mergedCE;
+    }
+
+    @Transactional(Transactional.TxType.REQUIRES_NEW)
+    public ConnectorEntity successfullyCalledService(ConnectorEntity connectorEntity, R serviceResponse) {
+        ConnectorEntity updatedEntity = updateEntityForSuccess(connectorEntity, serviceResponse);
+        connectorEntity.setWorkerId(null);
+        connectorEntity.setModifiedAt(ZonedDateTime.now(ZoneOffset.UTC));
+
+        LOGGER.info("Worker {} successfully called service with response : {} on entity: {} ", workerName(), serviceResponse, updatedEntity);
```

Comment:
Is it safe to log complete response and entity? It may have sensitive informations.

Reply:
I don't think it's much more sensitive than the actual OpenBridge data such as Processor ID, so I'll leave it now as it's useful for debugging